### ISSUE TITLE
Invoice error on Magento 2.1 v

### ISSUE THREAD

**Issue by nmallepally** [magento/magento2#5843]
Hello, 

I am using product community 2.1 v, the invoices does not work.
i know there are lot of issues reported on this but, its appearing on product addition.
we are using this version for production. can some one suggest the fix for it

does not allow me to submit invoice.
![invoice error](https://cloud.githubusercontent.com/assets/19605497/17181508/4b0d7e48-5418-11e6-8f9a-84512f55939f.png)

**Comment by dthampy** [comment]
@nmallepally ,

Couple of questions.

(i) Did you upgrade to CE 2.1.0? if so what was the upgrade path?
(ii) Do you have sample data installed with it?
(iii) Also can you details the steps to reproduce the issue? 

I can't seem to reproduce the problem in the screenshot in CE version 2.1.0.

**Comment by hostep** [comment]
This is a really well known issue in version 2.1.0 (@dthampy: why aren't you aware of this ?)

Is already fixed on the develop branch: https://github.com/magento/magento2/issues/4785

We are eagerly awaiting version 2.1.1 which will most likely fix this, but we have been waiting on this for over a month now ...

For a quick fix, see: https://github.com/magento/magento2/issues/5324

**Comment by nmallepally** [comment]
@dthampy its an upgrade from 2.0.7 to 2.1.
@hostep , after running the query it fixed the invoice issue, but i'm unable to pay using paypal.
after going to paypal payment gateway, and then clicking on continue, it redirects to website and gives me an error,  cannot place order.
![paypal 1](https://cloud.githubusercontent.com/assets/19605497/17205493/436d98a4-54a3-11e6-9365-a3f7af54690d.png)
![paypal 2](https://cloud.githubusercontent.com/assets/19605497/17205500/478f8d34-54a3-11e6-81aa-f00bf2a7f197.png)
![paypal 3 error](https://cloud.githubusercontent.com/assets/19605497/17205503/4c3028b2-54a3-11e6-9e51-d7a559324dd5.png)

**Comment by quienti** [comment]
@nmallepally , do you have any error in system.log or exception.log, or maybe a crash report ?

**Comment by Imaginaerum** [comment]
Can you set language to english. Flush all caches. Disable caches... And try PayPal order ?

**Comment by nmallepally** [comment]
@quienti this is what it showed on exception log,

] main.CRITICAL: Exception: Wrong order ID: "000000129". in /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-paypal/Model/Ipn.php:140
Stack trace:
#0 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-paypal/Model/Ipn.php(106): Magento\Paypal\Model\Ipn->_getOrder()
#1 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-paypal/Model/Ipn.php(87): Magento\Paypal\Model\Ipn->_getConfig()
#2 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-paypal/Controller/Ipn/Index.php(56): Magento\Paypal\Model\Ipn->processIpnRequest()
#3 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Paypal/Controller/Ipn/Index/Interceptor.php(24): Magento\Paypal\Controller\Ipn\Index->execute()
#4 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/App/Action/Action.php(102): Magento\Paypal\Controller\Ipn\Index\Interceptor->execute()
#5 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#6 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Paypal\Controller\Ipn\Index\Interceptor->___callParent('dispatch', Array)
#7 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'contextPlugin')
#8 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-store/App/Action/Plugin/Context.php(106): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#9 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\Context->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#10 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'customer-app-ac...')
#11 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-customer/Model/App/Action/ContextPlugin.php(61): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#12 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Customer\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#13 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'storeCheck')
#14 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-store/App/Action/Plugin/StoreCheck.php(44): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#15 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\StoreCheck->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#16 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'weee-app-action...')
#17 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-weee/Model/App/Action/ContextPlugin.php(112): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#18 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Weee\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#19 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'tax-app-action-...')
#20 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-tax/Model/App/Action/ContextPlugin.php(91): Magento\Paypal\Controller\Ipn\Index\Interceptor->Magento\Framework\Interception{closure}(Object(Magento\Framework\App\Request\Http))
#21 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Tax\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#22 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Paypal/Controller/Ipn/Index/Interceptor.php(39): Magento\Paypal\Controller\Ipn\Index\Interceptor->___callPlugins('dispatch', Array, Array)
#23 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/App/FrontController.php(55): Magento\Paypal\Controller\Ipn\Index\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#24 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#25 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#26 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'requestPreproce...')
#27 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php(94): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#28 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#29 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#30 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#31 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#32 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#33 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-page-cache/Model/App/FrontController/VarnishPlugin.php(55): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#34 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\PageCache\Model\App\FrontController\VarnishPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#35 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#36 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(68): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception{closure}(Object(Magento\Framework\App\Request\Http))
#37 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#38 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#39 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#40 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(146): Magento\Framework\App\Http->launch()
#41 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Framework/App/Http/Interceptor.php(26): Magento\Framework\App\Http\Interceptor->___callPlugins('launch', Array, Array)
#42 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http\Interceptor->launch()
#43 /var/www/vhosts/edwia.eu/httpdocs/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#44 {main} [] []

System log
#1 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-sales/Model/Order/Payment.php(359): array_key_exists(true, Array)
#2 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Sales/Model/Order/Payment/Interceptor.php(167): Magento\Sales\Model\Order\Payment->place()
#3 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-sales/Model/Order.php(871): Magento\Sales\Model\Order\Payment\Interceptor->place()
#4 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-sales/Model/Order.php(1060): Magento\Sales\Model\Order->_placePayment()
#5 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Sales/Model/Order/Interceptor.php(479): Magento\Sales\Model\Order->place()
#6 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-sales/Model/Service/OrderService.php(190): Magento\Sales\Model\Order\Interceptor->place()
#7 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-quote/Model/QuoteManagement.php(486): Magento\Sales\Model\Service\OrderService->place(Object(Magento\Sales\Model\Order\Interceptor))
#8 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-quote/Model/QuoteManagement.php(389): Magento\Quote\Model\QuoteManagement->submitQuote(Object(Magento\Quote\Model\Quote), Array)
#9 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-paypal/Model/Express/Checkout.php(792): Magento\Quote\Model\QuoteManagement->submit(Object(Magento\Quote\Model\Quote))
#10 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-paypal/Controller/Express/AbstractExpress/PlaceOrder.php(81): Magento\Paypal\Model\Express\Checkout->place('EC-56J070287424...')
#11 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Paypal/Controller/Express/PlaceOrder/Interceptor.php(24): Magento\Paypal\Controller\Express\AbstractExpress\PlaceOrder->execute()
#12 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/App/Action/Action.php(102): Magento\Paypal\Controller\Express\PlaceOrder\Interceptor->execute()
#13 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#14 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Paypal\Controller\Express\PlaceOrder\Interceptor->___callParent('dispatch', Array)
#15 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Array, 'contextPlugin')
#16 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-store/App/Action/Plugin/Context.php(106): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#17 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\Context->aroundDispatch(Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#18 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Array, 'customer-app-ac...')
#19 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-customer/Model/App/Action/ContextPlugin.php(61): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#20 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Customer\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#21 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Array, 'storeCheck')
#22 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-store/App/Action/Plugin/StoreCheck.php(44): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#23 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\StoreCheck->aroundDispatch(Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#24 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Array, 'weee-app-action...')
#25 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-weee/Model/App/Action/ContextPlugin.php(112): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#26 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Weee\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#27 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Paypal\...', 'dispatch', Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Array, 'tax-app-action-...')
#28 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-tax/Model/App/Action/ContextPlugin.php(91): Magento\Paypal\Controller\Express\PlaceOrder\Interceptor->Magento\Framework\Interception{closure}(Object(Magento\Framework\App\Request\Http))
#29 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Tax\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Express\PlaceOrder\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#30 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Paypal/Controller/Express/PlaceOrder/Interceptor.php(104): Magento\Paypal\Controller\Express\PlaceOrder\Interceptor->___callPlugins('dispatch', Array, Array)
#31 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/App/FrontController.php(55): Magento\Paypal\Controller\Express\PlaceOrder\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#32 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#33 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#34 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'requestPreproce...')
#35 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php(94): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#36 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#37 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#38 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#39 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#40 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#41 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-page-cache/Model/App/FrontController/VarnishPlugin.php(55): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain{closure}(Object(Magento\Framework\App\Request\Http))
#42 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\PageCache\Model\App\FrontController\VarnishPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#43 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#44 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(68): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception{closure}(Object(Magento\Framework\App\Request\Http))
#45 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#46 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#47 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#48 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/Interception/Interceptor.php(146): Magento\Framework\App\Http->launch()
#49 /var/www/vhosts/edwia.eu/httpdocs/var/generation/Magento/Framework/App/Http/Interceptor.php(26): Magento\Framework\App\Http\Interceptor->___callPlugins('launch', Array, Array)
#50 /var/www/vhosts/edwia.eu/httpdocs/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http\Interceptor->launch()
#51 /var/www/vhosts/edwia.eu/httpdocs/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#52 {main} [] []

**Comment by Imaginaerum** [comment]
@quienti Merci l'aide... Il faut voir si il s'agit de problèmes js car c'est une plaie... je vais voir pour remplacer les ' par un caractère spécial.

**Comment by Imaginaerum** [comment]
It seem there is a closure problem...

**Comment by nmallepally** [comment]
@Imaginaerum @quienti  i did try to change the locale to english and try to do the paypal, when i clicked on paypal, without going to the gateway, it returned me a 
![error paypal](https://cloud.githubusercontent.com/assets/19605497/17208628/6409c0ac-54b1-11e6-8fed-ff963325ba2e.png)
PayPal gateway has rejected request. Payment has already been made for this InvoiceID (#10412: Duplicate invoice).

**Comment by Imaginaerum** [comment]
@nmallepally This order already exists.
Please, create a new test order.
@quienti I create a translation test with no single quote. I've create an issue on french language pack

**Comment by quienti** [comment]
@nmallepally , yes, try with a new order
@Imaginaerum Okay, I will try to simulate an order with Paypal with french pack in the day.

**Comment by Imaginaerum** [comment]
@nmallepally @quienti can you try with this [Dictionary](https://raw.githubusercontent.com/Imaginaerum/magento2-language-fr-fr/master/dictionary.csv) ?

**Comment by nmallepally** [comment]
@Imaginaerum @quienti this is what i found
![paypal error 2 1](https://cloud.githubusercontent.com/assets/19605497/17208987/e626be2c-54b2-11e6-8fad-cac9bb660ccd.png)

**Comment by Imaginaerum** [comment]
@nmallepally same bug into english.

**Comment by nmallepally** [comment]
@Imaginaerum Yes,

we can't place the order

**Comment by quienti** [comment]
So, this is a bug from Magento, you need to wait a response from a Magento members, sorry @nmallepally

**Comment by nmallepally** [comment]
@quienti thanks for your time, so the errors from log above are normal? is there a way to fix them?

**Comment by quienti** [comment]
No, this is not normal, but this error can help Magento members to resolve this problem.

**Comment by nmallepally** [comment]
@quienti yeah i just found that there are plenty of similar issues being reported , there is no solution for it at the moment i suppose its better to stop paypal payments. i found its being fixed on MAGETWO-54787 internally.

**Comment by quienti** [comment]
Good, so if there are another issue like you, you can close this issue, and you can click on the button "subcribe" in this issue related, and when a person add comment on this issue, you are notified!

**Comment by nmallepally** [comment]
@quienti I'm not sure, where to track the fix as well, paypal is our main payment method.
we have less time to fix, before we go live. It works well with authipay payment method.
Paypal does not work, gives me this error every time.

**Comment by quienti** [comment]
@nmallepally , I understand, but I can not help you for this problem, you need to wait a response from Magento members :/

**Comment by nmallepally** [comment]
@quienti Yeah, sure. thanks.  I even did try to do the real payment, it returns me the same error.
will wait for the magento guys to come back. Thanks again for the support.

**Comment by antboiko** [comment]
Hi @nmallepally , let me inform you that the fix is already in develop branch and will be available in 2.1.1 patch.

Best,
Anton.

**Comment by nmallepally** [comment]
Hi @antboiko Can you give me the link to fix on develop branch, so that i can fix it on my website.

**Comment by antboiko** [comment]
Hi @nmallepally , please the following commits 65f263b2bc1d1af62235099007e45da5f1f732f8, 7f5b6443c8aad774bb3389fca9c4d372aaa3f1af, 4173292aab6d4f84dfbe61da0fd7a09765f0db6e, b110260645313f4a9c3749637e6f8a7b37a2fffb, 0fff31d5cac793b7f7dfc4788a8c4911973d26b6, 93402f5c3e992e74de87444b11224adb5c4fe395

Regards,
Anton.

**Comment by nmallepally** [comment]
Great. Thanks @antboiko  Thanks, will implement and let you know the output.

**Comment by nmallepally** [comment]
Does above fixes would also work for paypal payment action method-> order and authorization as well?.
@antboiko

**Comment by nmallepally** [comment]
where do i find the structure app/code/Magento/Sales/Setup/InstallSchema.php 
app/code/Magento/Sales/Setup/UpgradeSchema.php

should i do the above changes here httpdocs / vendor / magento / module-sales / Setup / InstallSchema.php
httpdocs / vendor / magento / module-sales / Setup / UpgradeSchema.php

**Comment by dthampy** [comment]
Depending on whether you did a composer install or a clone ,the modules are deployed in different location in the directory structure.

if you did a clone, they would be in app/code/Magento/module
However, if you install via composer, they would be in vendor/magento/module-xyz .
Hope this helps.

**Comment by nmallepally** [comment]
@dthampy Thank you.
@antboiko I have done the above commits, The paypal does not work for payment action ORDER, it works for sale and authorization type, when tested in real mode.
But three of them does not work during test mode.
Another strange error is that, It does not allow me to order cash on delivery or other payment methods. Gives me an  error unable to pay at this time only for a single customer account, it works when tested as a guest and other customer accounts.
IT also gives me duplicate invoice error, gets stuck with duplicate invoice error.
Paypal error appears on home page and product pages after closing and opening browser as well.
![error on home page](https://cloud.githubusercontent.com/assets/19605497/17401677/cbdc3a44-5a46-11e6-9117-8e25c5e2deab.png)
![duplicate invoice](https://cloud.githubusercontent.com/assets/19605497/17401682/d0b8d73e-5a46-11e6-8825-6bfab554204b.png)
![place an order error](https://cloud.githubusercontent.com/assets/19605497/17401693/e5d0db26-5a46-11e6-9894-7a20326e67a2.png)

**Comment by dthampy** [comment]
@nmallepally , created an internal jira ticket - MAGETWO-56760 for your issue

**Comment by sma09sjb** [comment]
Hi, 

Do we have an update on this, i'm experiencing a similar bug when i configure Paypal Express Checkout. The payment goes through in my sandbox account, but i'm getting the following critical exception

`Exception: Wrong order ID: "000000005".`

![image](https://cloud.githubusercontent.com/assets/19289786/17767596/7c1edc5e-6528-11e6-8bb6-4f753084d01b.png)

And when i return back to my site, i'm returned back to the review page, even though i have not enabled the review step. And the following error message is displayed

`We can't create the order` 

I'm also experiencing the same error with a stripe integration, the order isn't created in the backend. 

As payments are such a fundamental part of any ecommerce site, i'm disappointed there are seemingly so many issues in this area. 

When will the patch 2.1.1 be released, i've tried manually updating the core code in accordance with the revisions above, but this has not resolved the issue and this is becoming a serious problem for our client.

Any update on this would be greatly appreciated.

KInd regards,

Sophie

**Comment by sma09sjb** [comment]
Any update on this? Could this be to do with setting up a test payment account and configuring it on two environments, i.e. having two potential order streams, within Paypal. I've removed the configured details from all but 1 environment, but still having issues with both Paypal and Stripe. Any help, or an estimate of how long I will need to wait before a fix is found, would be appreciated.

**Comment by vishveskrish** [comment]
@nmallepally : Do you find the solution regards to PayPal express ? We are facing same issue but no luck.

**Comment by nmallepally** [comment]
@vishveskrish It works for me on Live environment, for payment action type - sale and authorization on above commits.
Does not work for orders though. if you are testing in sandbox mode, try switching to live by changing payment action and give a try. may be it should work out.
It isn't  a good solution, but we had to to move with something working on paypal as we cannot block go live.

**Comment by vishveskrish** [comment]
@nmallepally : Have You  run the below query to resolve the issue ?

ALTER TABLE sales_invoice_grid ADD base_grand_total decimal(12,4) AFTER grand_total;

Or what steps you have performed can you please list down ?

Thank you!

**Comment by oceancloud82** [comment]
2.1.8 still have this issue.
[2017-08-30 18:10:00] main.CRITICAL: Exception: Wrong order ID: "XXXXXXXXXX". in /opt/bitnami/apps/magento/htdocs/vendor/magento/module-paypal/Model/Ipn.php:140
Stack trace:
#0 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-paypal/Model/Ipn.php(106): Magento\Paypal\Model\Ipn->_getOrder()
#1 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-paypal/Model/Ipn.php(87): Magento\Paypal\Model\Ipn->_getConfig()
#2 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-paypal/Controller/Ipn/Index.php(56): Magento\Paypal\Model\Ipn->processIpnRequest()
#3 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/App/Action/Action.php(102): Magento\Paypal\Controller\Ipn\Index->execute()
#4 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#5 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Paypal\Controller\Ipn\Index\Interceptor->___callParent('dispatch', Array)
#6 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'contextPlugin')
#7 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-store/App/Action/Plugin/Context.php(106): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#8 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\Context->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#9 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'customer-app-ac...')
#10 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-customer/Model/App/Action/ContextPlugin.php(61): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#11 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Customer\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#12 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'storeCheck')
#13 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-store/App/Action/Plugin/StoreCheck.php(44): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#14 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\StoreCheck->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#15 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'weee-app-action...')
#16 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-weee/Model/App/Action/ContextPlugin.php(112): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#17 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Weee\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#18 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'tax-app-action-...')
#19 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-tax/Model/App/Action/ContextPlugin.php(91): Magento\Paypal\Controller\Ipn\Index\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#20 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Tax\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#21 /opt/bitnami/apps/magento/htdocs/var/generation/Magento/Paypal/Controller/Ipn/Index/Interceptor.php(26): Magento\Paypal\Controller\Ipn\Index\Interceptor->___callPlugins('dispatch', Array, Array)
#22 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/App/FrontController.php(55): Magento\Paypal\Controller\Ipn\Index\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#23 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#24 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#25 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'requestPreproce...')
#26 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php(94): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#27 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#28 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#29 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#30 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#31 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#32 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-page-cache/Model/App/FrontController/VarnishPlugin.php(55): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#33 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\PageCache\Model\App\FrontController\VarnishPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#34 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#35 /opt/bitnami/apps/magento/htdocs/vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(73): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#36 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#37 /opt/bitnami/apps/magento/htdocs/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#38 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#39 /opt/bitnami/apps/magento/htdocs/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#40 /opt/bitnami/apps/magento/htdocs/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#41 {main} [] []

**Comment by maderlock** [comment]
I can confirm that I'm seeing this in 2.1.9.

Is there a workaround for this a.t.m?

**Comment by magento-engcom-team** [comment]
@nmallepally, thank you for your report.
We've created internal ticket(s) MAGETWO-56760 to track progress on the issue.

**Comment by zack6849** [comment]
We've also gotten this error on 2.1.9, the invoice failed to turn into an order and returns 500 internal server error every time.

**Comment by s00071609** [comment]
This issue has been there for years. Upgraded to 2.2.2, the error still occurs. Probably the most ignored issue that kills the checkout process. Its really sad

**Comment by ishakhsuvarov** [comment]
Hi @nmallepally, @s00071609, @zack6849, @maderlock 
Fix for the issue should be delivered with the 2.2.4 release with the commit 137dc666d76cc262f435cae22cd7f0f8759999ee

**Comment by magento-engcom-team** [comment]
Closing, as the fix is delivered to the `2.2-develop` branch.
